src/assignments/orbit-report.rst

```diff
@@ -0,0 +1,414 @@
+
+.. sourcecode:: typescript
+   :linenos:
+
+   sort(column: string): void {
+    // array.sort modifies the array, sorting the items based on the given compare function
+    this.satellites.sort(function(a: Satellite, b: Satellite): number {
+      if(a[column] < b[column]) {
+        return -1;
+      } else if (a[column] > b[column]) {
+        return 1;
+      }
+      return 0;
+     });
+   }
+
+
+8) Searching
+^^^^^^^^^^^^
+You are doing great! Only two more features to add. Next you will add a search feature.
+
+#. Add this HTML ``<div class="search-form"></div>`` in your ``app.component.html``.
+#. Add an ``<input>`` element insde the ``<div>``.
+#. Add a ``<button>`` element insde the ``<div>``.
+#. Add an Angular click handler to the ``<button>`` that when clicked calls ``search(searchTerm.value)``
+
+   * ``searchTerm`` being the local variable name for the ``<input>``
+
+#. Add an Angular keyup.enter handler to the ``<input>`` that when clicked calls ``search(searchTerm.value)``
+
+   * ``searchTerm`` being the local variable name for the ``<input>``
+
+#. Add a ``search`` method to the ``AppComponent`` class.
+
+   * The ``search`` method is provded below.
+
+.. sourcecode:: typescript
+   :linenos:
+
+   search(searchTerm: string): void {
+      let matchingSatellites: Satellite[] = [];
+      searchTerm = searchTerm.toLowerCase();
+      for(let i=0; i < this.sourceList.length; i++) {
+         let name = this.sourceList[i].name.toLowerCase();
+         if (name.indexOf(searchTerm) >= 0) {
+            matchingSatellites.push(this.sourceList[i]);
+         }
+      }
+      // assign this.displayList to be the the array of matching satellites
+      // this will cause Angular to re-make the table, but now only containg matches
+      this.displayList = matchingSatellites;
+   }
+
+Notice the usage of a new variable named ``displayList``. ``displayList`` should contain the ``Satellite`` objects
+that the user wants to see. That was previously ALL the satellites. Now the user can perform a search, which
+means they want to see the matching results. ``sourceList`` will contain ALL the ``Satellite`` objects. If you removed
+the ``Satellite`` objects from ``sourceList``, then the user could never see them again. Instead when the user
+performs a search, ``displayList`` will be populated with the all the matching ``Satellite`` objects in ``sourceList``.
+Matching is defined as ``satellite.name`` containing the search term.
+
+7. Add the ``displayList: Satellite[];`` property to the ``AppComponent`` class.
+
+   * Set ``displayList = []`` in the constructor.
+
+8. Pass in the ``displayList`` to the ``orbit-list-component``.
+
+   * ``<app-orbit-list [satellites]="displayList"></app-orbit-list>``
+
+9. For an example of search working, see :ref:`Orbit Report Demo <orbit-report-demo>`.
+
+
+9) Counting Satellites
+^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
This whole step needs to be reworked.

Reply:
Do you have any suggestions on what should be changed?

Reply:
I can see what you mean about letting students figure out things on their own, but with how much support they receive on the other steps, it felt like a harsh drop-off as opposed to an inspiring push towards independence. I would reference the other steps where they did those things! Especially since people are going to be working on the assignment for multiple days.

Reply:
I appreciate your concerns and comments, but I am expected the students to be able to refer to previous steps and also previous chapters to help them accomplish this task.